.github/workflows/build.yml

```diff
@@ -134,35 +142,51 @@ jobs:
           version: ${{ env.version }}
           target: release-default
           arch: ${{ matrix.arch }}
-          # The ECR tag used below is to allow us to test by pushing to a private ECR registry.
-          # If you want to pubish to ECR use the following syntax instead - public.ecr.aws/hashicorp/${{env.repo}}:${{env.version}}.
-          # Please note - if you haven't previously been publishing to ECR you will need to reach out to team-rel-eng
-          # - https://hashicorp.atlassian.net/wiki/spaces/RDXPOC/pages/2298218311/How+to+Push+a+Docker+image+to+ECR
           tags: |
             docker.io/hashicorp/${{ env.repo }}:${{ env.version }}
             public.ecr.aws/hashicorp/${{ env.repo }}:${{ env.version }}
-          # dev_tags are tags that get automatically pushed whenever successful
-          # builds make it to the stable channel. The intention is for these tags
-          # to be used for early testing of new code prior to official releases
-          # going out. The stable channel implies that all tests and scans have
-          # completed successfully, so these images should be _stable_ but are not
-          # intended for production use.
-          #
-          # Here we have two example dev tags. The first (ending -dev) is a tag
-          # that will be updated over-and-over as new builds arrive in stable.
-          #
-          # The second (using the git SHA) will produce a new separate tag for
-          # each commit that is built. (These can still be overridden if the same
-          # commit is built successfully a second time, but that is a less likely
-          # scenario.) These kinds of dev tags are useful if you want to be able
-          # to use Docker images built from those specific commits.
-          #
-          # NOTE: dev_tags MUST publish to the 'hashicorppreview' DockerHub org, it
-          # will fail to any other DockerHub org or registry. You can optionally
-          # prepend docker.io
           dev_tags: |
             hashicorppreview/${{ env.repo }}:${{ env.version }}
-            hashicorppreview/${{ env.repo }}:${{ env.version }}-${{ github.sha }}
+            docker.io/hashicorppreview/${{ env.repo }}:${{ env.version }}-${{ github.sha }}
+
+  build-docker-fips:
+    name: Docker ${{ matrix.arch }} FIPS release build
+    needs:
+      - get-product-version
+      - build-linux
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - { arch: "arm64" }
+          - { arch: "amd64" }
+    env:
+      repo: ${{ github.event.repository.name }}
+      version: ${{ needs.get-product-version.outputs.product-version }}+fips1402
+
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - name: Docker FIPS Build (Action)
+        uses: hashicorp/actions-docker-build@v1
```

Comment:
The version for `actions-docker-build` should be pinned to a SHA.

Reply:
I think the SHA pinning is only for external actions, that way CRT can more gracefully make small changes with our homegrown actions - at least that is how it appears to be in the other Consul repos